tests/end_to_end/smoke_test.py

```diff
@@ -5,40 +5,44 @@
 import os
 
 SCRIPT = """
-source {envloc}/bin/activate;
+source {envloc}/{folder}/activate;
+python ok
 python ok {args} > {stdoutloc} 2> {stderrloc}
 """
 
 class SmokeTest(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
-        cls.clean_env_loc = tempfile.TemporaryDirectory().name
+        cls.clean_env_loc = tempfile.TemporaryDirectory().name.replace("\\", "/")
         cls.create_clean_env()
 
     @classmethod
     def create_clean_env(cls):
-        subprocess.check_call(["virtualenv", "-q", "-p", "python", cls.clean_env_loc])
+        subprocess.check_call(["virtualenv", "-p", "python", cls.clean_env_loc])
 
     def setUp(self):
-        self.directory = tempfile.TemporaryDirectory().name
+        self.directory = tempfile.TemporaryDirectory().name.replace("\\", "/")
         publish.package_client(self.directory)
 
     def run_ok(self, *args):
-        out_loc = tempfile.NamedTemporaryFile().name
-        err_loc = tempfile.NamedTemporaryFile().name
+        _, out_loc = tempfile.mkstemp()
+        _, err_loc = tempfile.mkstemp()
         command_line = SCRIPT.format(
             envloc=self.clean_env_loc,
+            folder="scripts" if os.name == "nt" else "bin",
             args=" ".join(args),
-            stdoutloc=out_loc,
-            stderrloc=err_loc,
+            stdoutloc=out_loc.replace("\\", "/"),
+            stderrloc=err_loc.replace("\\", "/"),
         )
-        subprocess.call(command_line, shell=True, cwd=self.directory, executable='/bin/bash')
+        subprocess.check_call(command_line, shell=True, executable=os.environ['SHELL'], cwd=self.directory)
```

Comment:
I don't think you can this. `shell` in a pure-Windows version of Python (which is what I think you have) means the Command Prompt, not whatever shell you happen to have invoked Python from.

If you want to run it in Bash, it's not trivial (it might have a pitfall or two). Ping me when you're done and I can help you figure it out.

Reply:
Actually, you can probably use pass the script on stdin to `sh` without making a mess:

    subprocess.check_output(os.getenv('SHELL', 'sh'), input=SCRIPT.encode('utf-8'))

Don't use `subprocess.check_call` since it doesn't support `input`.  
Also don't use `bash` instead of `sh`, since that might potentially call into WSL instead of failing if the shell isn't present.

Reply:
ok trying this out let's see if it works

Reply:
thanks that fixed it!